src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -1,171 +1,269 @@
-                return new AddTaskCommand(title, dayOfTask, information);
-            } else {
-                // no information entered
-                dayOfTask = addInfo.substring(positionOfDay + SEPARATING_SPACE).strip();
-                return new AddTaskCommand(title, dayOfTask, "");
-            }
-        case LESSON_KEYWORD:
-            int positionOfStartTime = addInfo.indexOf(FLAG_START_TIME);
-            int positionOfEndTime = addInfo.indexOf(FLAG_END_TIME);
-            String dayOfLesson = addInfo.substring(positionOfDay + SEPARATING_SPACE, positionOfStartTime - 1).strip();
-            String lessonStartTime =
-                    addInfo.substring(positionOfStartTime + SEPARATING_SPACE, positionOfEndTime - 1).strip();
-            String lessonEndTime = addInfo.substring(positionOfEndTime + SEPARATING_SPACE).strip();
-            return new AddLessonCommand(title, dayOfLesson, lessonStartTime, lessonEndTime);
+    private static Command parseAddCommand(String userResponse) throws DukeException {
+        String param = userResponse.replaceFirst("add", "").strip();
+        CommandType commandType = getCommandType(param);
+
+        switch (commandType) {
+        case TASK:
+            return parseAddTaskCommand(param.replaceFirst("task", "").strip());
+        case LESSON:
+            return parseAddLessonCommand(param.replaceFirst("lesson", "").strip());
+        case INVALID:
+            // Fallthrough
         default:
-            throw new DukeException(Message.ERROR_NO_ADD_TYPE);
+            throw new DukeException(Message.ERROR_INVALID_COMMAND);
         }
     }
 
-    /**
-     * Returns an executable Command object containing information about the type of tasks or lessons to list, and
-     * the time period of items to list.
-     *
-     * @param listInfo the remainder of the command following the word "list"
-     * @return the Command to be executed to list items
-     */
-    private Command parseListCommand(String listInfo) {
-        int positionOfSpace = listInfo.indexOf(" ");
-        String listType = listInfo.substring(0, positionOfSpace); // either all, task or lesson
-        String period = listInfo.substring(positionOfSpace + 1); // today, week, tomorrow, mon, tue, wed, thu, fri,
-        // sat, or sun
+    private static Command parseAddTaskCommand(String userResponse) throws DukeException {
+        String[] params = userResponse.split(" -d | -i ");
+        if (params.length < 2 || params.length > 3) {
+            throw new DukeException(Message.ERROR_INVALID_COMMAND);
+        }
 
-        return new ListCommand(listType, period);
+        String title = params[0].strip();
+        String dayOfTheWeek = params[1].strip();    // TODO: Validate correctness with enum
+
+        switch (params.length) {
+        case 2:
+            // System.out.print(title + ", " + dayOfTheWeek);      // TODO: to be removed

```

Comment:
Just to clarify, what is the reason that this commented out code is kept instead of straight up removing it?